CMakeLists.txt

```diff
@@ -98,8 +130,19 @@ elseif(BUILD_COMPILER STREQUAL "msvc")
     add_compile_options(/MP) # Multi-threaded build
     add_compile_options(/Zc:preprocessor) # Use standard compliant preprocessor
     add_compile_options(/we4834) # Discarding return value of function with 'nodiscard' attribute
-    add_link_options(/LARGEADDRESSAWARE) # Enable heap size over 2gb for x86 builds
-    add_link_options(/SAFESEH:NO) # This is for x86 build
+    if (CHECK_CPU_ARCHITECTURE_X86)
+        add_link_options(/LARGEADDRESSAWARE) # Enable heap size over 2gb for x86 builds
+        add_link_options(/SAFESEH:NO) # This is for x86 build
+        add_link_options(/OPT:REF) # in 32-bit environments, /OPT:REF is recommended
+    else()
+        add_link_options(/OPT:REF,ICF) 
+    endif()
+    # Allows the compiler to package individual functions in the form of packaged functions (COMDATs).
+    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} /Gy")
+    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /Gy")
```

Comment:
Why not `add_compile_options`?

Reply:
We only want /Gy on release builds, add_compile_options adds it for debug as well

Reply:
Use a generator expression?